python/ray/serve/autoscaling_policy.py

```diff
@@ -0,0 +1,131 @@
+from abc import ABCMeta, abstractmethod
+
+from ray.serve.utils import logger
+
+
+class AutoscalingPolicy:
+    """Defines the interface for an autoscaling policy.
+
+    To add a new autoscaling policy, a class should be defined that provides
+    this interface. The class may be stateful, in which case it may also want
+    to provide a non-default constructor. However, this state will be lost when
+    the controller recovers from a failure.
+    """
+    __metaclass__ = ABCMeta
+
+    def __init__(self, config):
+        """Initialize the policy using the specified config dictionary."""
+        self.config = config
+
+    @abstractmethod
+    def scale(self, router_queue_lens, curr_replicas):
+        """Make a decision to scale backends.
+
+        Arguments:
+            router_queue_lens: Dict[str: int] mapping routers to their most
+                recent queue length for this backend.
+            curr_replicas: int The number of replicas that the backend
+                currently has.
+
+        Returns:
+            int The new number of replicas to scale this backend to. Returns -1
+                if there should be no change.
```

Comment:
Why does the policy need to return -1? If there should be no change, the policy should just return `curr_replicas`.